File: tests/java/tests/java/org/elastos/hive/config/CrossConfig.java
package org.elastos.hive.config;

import com.fasterxml.jackson.annotation.JsonProperty;

public class CrossConfig extends Config {
	@JsonProperty("application")
	private ApplicationConfig applicationConfig;
	@JsonProperty("user")
	private UserConfig userConfig;
	@JsonProperty("crossDid")
	private String crossDid;

	public ApplicationConfig applicationConfig() {
		return this.applicationConfig;
	}

	public UserConfig userConfig() {
		return this.userConfig;
	}

	public String crossDid() {
		return this.crossDid;
	}

	public static CrossConfig deserialize(String content) {
		return deserialize(content, CrossConfig.class);
	}
}

File: tests/src/config/crossconfig.ts
import { ApplicationConfig } from "./applicationconfig"
import { UserConfig } from "./userconfig"

export class CrossConfig {
	private application: ApplicationConfig;
	private user: UserConfig;
	private crossDid: string;

	public getApplicationConfig(): ApplicationConfig {
		return this.application;
	}

	public getUserConfig(): UserConfig {
		return this.user;
	}

	public getCrossDid(): string {
		return this.crossDid;
	}

	public static deserialize(json: string): CrossConfig {
		let jsonObj = JSON.parse(json);
		let newConfig = new CrossConfig();

		newConfig.application = ApplicationConfig.deserialize(jsonObj['application']);
		newConfig.user = UserConfig.deserialize(jsonObj['user']);
		newConfig.crossDid = jsonObj['crossDid'];

		return newConfig;
	}
}
